### ISSUE TITLE
Switch `c10::optional<ShapeSymbol>` with just `ShapeSymbol`.

### ISSUE THREAD

**Issue by Krovatkin** [pytorch/pytorch#38411]
## 🚀 Feature
* Simplify the `symbolic_sizes_` property of TensorType API. `TensorType` is a class to represent a tensor type in TorchScript IR. `TensorType` tracks various properties of a tensor such as tensor shape, strides, device, requires_grad, etc.  TorchScript IR consists mostly of `Node` and `Value` . `Node`s denote operations that produce outputs (`Value`s) . Each value is statically typed with `Type` and that type can be `TensorType` .
*  Currently, an unknown size is expressed with [VaryingShape](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L423) that wraps around a vector of `c10::optional<int64_t>`s
* We recently added `ShapeSymbol` class to represent `size` and values and it has its own way of expressing unknown values`
    * ShapeSymbol now provides a special symbol (`-1`) which we can use to denote unknown symbols (https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L389) 
 * In short, we would like to switch from using optionals to denote unknown symbols to using that special symbol.
## Motivation

* The outcomes we hope for
    * The reduction in the amount of boxing/unboxing
    * Avoiding having more than one way to express the same thing.

## Pitch
* We would like to add `unknownSymbol` to denote unknown/varying shape
* We would like to use `unknownSymbol` in lieu of `c10::optional<ShapeSymbol>`
* We would like to replace `VaryingShape<ShapeSymbol>` with just `c10::optional<std::vector<ShapeSymbol>>` 

## Alternatives

Do nothing n and suffer from the current design.

## Additional context

For questions, ping @Krovatkin 

cc @suo

**Comment by Krovatkin** [comment]
@suo Please reassign accordingly since I don't know yanan's GH handle.
